Docs/development/building.md

```diff
@@ -62,7 +62,11 @@ rake reload:release
 
 #### Debugging
 
-Xcode and lldb can be used to debug Santa, just like any other project. Instead of clicking the play button to launch and attach to a process, you can attach to an already running, or soon to by running, component of Santa. To do this select the Debug menu and choose `Attach to Process by PID or Name… ` . Below are the four components of Santa and who to debug the process as. Note: santa-driver (the kernel extension) cannot be debugged by attaching with Xcode.
+Xcode and lldb can be used to debug Santa, just like any other project. Instead of clicking the play button to launch and attach to a process, you can attach to an already running, or soon to by running, component of Santa. To do this select the Debug menu and choose `Attach to Process by PID or Name… `. Below are the four components of Santa and who to debug the process as. 
+
+Note: santa-driver (the kernel extension) cannot be debugged by attaching with Xcode.
+
+Note: Attaching to santad with breakpoints in the decision making codepath will deadlock your machine.
```

Comment:
Not even setting breakpoints, just attaching will deadlock. Attaching 'interrupts' the process, stopping all threads from proceeding and dropping you into interactive mode, but now now thing is processing, the terminal will freeze and you won't be able to resume.